#! /usr/bin/env python
# check out https://wiki.ros.org/dynamic_reconfigure/Tutorials/HowToWriteYourFirstCfgFile

PACKAGE="mcu_interface"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       Name               Type,   Reconfig.lvl            Description                       Default   Min   Max
#gen.add("velocity_scale_factor",  double_t, 0,         "Maximum velocity scale factor",       0.5,      0.0,  1.0)
#gen.add("number_persons",          int_t,   0,         "number of targets",                     0,      0,    5000)
#gen.add("force_map_path",          str_t,   0,         "path to force map file",            "/home")
gen.add("start_sync",              bool_t,   0,          "When set to True, synchornizes MCU and Smartphone",      False)
exit(gen.generate(PACKAGE, "mcu_interface", "parameters"))
# last parameter is a name prefix the generated files will get (e.g. "<name>Config.h for cpp
